### PR TITLE
Use latest compose specs

### PR DESCRIPTION
**Pull request by lucasfcnunes**
## What type of PR is this? 

- [x] Refactor

## Description

Commits to improve/refactor/update the Redash project to current docker specs.

## How is this tested?

- [x] Manually

`make build` runs as always.

## Related Tickets & Documents

- https://docs.docker.com/compose/compose-application-model/#the-compose-file (why `docker-compose.yml` -> `compose.yaml`)
- https://github.com/compose-spec/compose-spec/issues/352
- https://www.docker.com/blog/announcing-compose-v2-general-availability/
- https://docs.docker.com/compose/compose-file/#compose-file

### CODE REVIEW

**Comment by lucasfcnunes** [comment]
Build failing because of CircleCI's docker [compose] version (v1)?
```
Building server
Sending build context to Docker daemon  14.15MB
Error response from daemon: Dockerfile parse error line 12: Unknown flag: mount
ERROR: Service 'server' failed to build : Build failed
```

**Comment by guidopetri** [comment]
Hi @lucasfcnunes , thanks for your contribution! We've updated a lot of things now that we're community-driven so would you mind rebasing off `master` to re-run the CI, as well as updating merge conflicts?

Additionally, if you could split up the build caching changes into a separate PR, that would be ideal - that way we can keep the required changes to comply with recommendations separate from complex Dockerfile changes.

File: Dockerfile [hunk 1]
```diff
@@ -1,31 +1,48 @@
...
+  echo "Downloading SimbaSparkODBC..." \
+  && wget -O - "$databricks_odbc_driver_url" | busybox unzip -j - "*/simbaspark*.deb" -d . \
+  echo "Done!"
+
+FROM node:14.17 AS frontend-builder
 
-RUN npm install --global --force yarn@1.22.10
+RUN \
+  --mount=type=cache,target=/root/.npm \
+  npm install --global --force yarn@1.22.10
```
Comment (justinclift) [inline Dockerfile:14]: As a data point, we've updated to yarn@1.22.19 a few weeks ago (241dcfacd98092be4ae2830d33b69e185f12a91a), so probably best to use that version here too.

**Comment by justinclift** [comment]
Looking over this PR, it's probably going to need very manual re-creation as we've been changing so many things in the Dockerfile over the last few weeks.

Not impossible in any way, but likely just pretty time consuming.

**Comment by guidopetri** [comment]
Agreed, that's why I think it's better splitting up into more "style" and more "feature" type changes.

**Comment by justinclift** [comment]
Good thinking. :smile:

**Comment by guidopetri** [comment]
@lucasfcnunes thanks for updating! I'm not a huge fan personally of changing the docker compose filename, but I see it's the preferred standard now (why change what isn't broken?). It seems we need to update something else on the CI if we want to rename the files though, otherwise the CI won't pass.

**Comment by lucasfcnunes** [comment]
> @lucasfcnunes thanks for updating! I'm not a huge fan personally of changing the docker compose filename, but I see it's the preferred standard now (why change what isn't broken?). It seems we need to update something else on the CI if we want to rename the files though, otherwise the CI won't pass.

@guidopetri I think it's expected as it's using the `ci.yml` from `master` branch, not sure.
*These [lines](https://github.com/getredash/redash/pull/6716/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fL6-R8) prevent [this](https://nathandavison.com/blog/github-actions-and-the-threat-of-malicious-pull-requests#:~:text=What%20this%20means%20is%2C%20the%20pull_request_target%20event%20does%20not%20use%20anything%20from%20the%20fork%20PR%20at%20all%20when%20executing%20the%20workflow%20configured%20against%20that%20action%2C%20it%20simply%20executes%20the%20(presumably%20safe)%20code%20and%20configuration%20already%20in%20the%20base%20repo.).

image

**Comment by guidopetri** [comment]
Ah, I see, that's new to me! Thanks for explaining. Let's go ahead and merge then, and hopefully it passes in master. Thanks again for your contribution!

**Comment by lucasfcnunes** [comment]
Seems to be working as expected https://github.com/getredash/redash/actions/runs/7777381717/workflow#L33

What I don't understand is how this PR was "merged" into master but is still open...
image

**Comment by guidopetri** [comment]
Me neither, I think it was a github bug. I'm going to close it, but thanks again!

**Pull request closed without merging**